### PR TITLE
Python: Embeddings batch size

### PR DESCRIPTION
**Pull request by mkarle**
### Motivation and Context

Adding optional batch size parameter for when Azure OpenAI limits number of inputs for embeddings requests. Fixes https://github.com/microsoft/semantic-kernel/issues/2099 
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py [hunk 1]
```diff
@@ -52,26 +52,31 @@ def __init__(
         self._org_id = org_id
         self._log = log if log is not None else NullLogger()
 
-    async def generate_embeddings_async(self, texts: List[str]) -> ndarray:
+    async def generate_embeddings_async(
+        self, texts: List[str], batch_size: int = 0
```
Comment (awharrison-28) [inline python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py:56]: default batch size should be 1

```suggestion
        self, texts: List[str], batch_size: int = 1
```

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py [hunk 2]
```diff
@@ -52,26 +52,31 @@ def __init__(
...
-                api_base=self._endpoint,
-                api_version=self._api_version,
-                organization=self._org_id,
-                input=texts,
-            )
-
-            # make numpy arrays from the response
-            raw_embeddings = [array(x["embedding"]) for x in response["data"]]
+            raw_embeddings = []
+            batch_size = batch_size or len(texts)
```
Comment (awharrison-28) [inline python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py:66]: if len(texts) > batch_size, (say 2 and 1) the batch_size will result in 2 which could cause unexpected behavior. 

```suggestion
            batch_size = min(batch_size, len(texts)
```

**Review by awharrison-28 — CHANGES_REQUESTED** [review]

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py (hunk 1 above)
Comment (mkarle) [reply python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py]: Made it Optional[int] = None

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py (hunk 2 above)
Comment (mkarle) [reply python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_embedding.py:66]: Talked offline, this isn't the case

**Review by awharrison-28 — APPROVED** [review]

**Pull request merged**